packages/tencent_cloud/data_stream/audit/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,195 @@
+---
+description: Pipeline to process tencent_cloud.audit logs
+processors:
+  - set:
+      tag: set_ecs_version
+      field: ecs.version
+      value: 8.17.0
+  - rename:
+      field: message
+      tag: rename_message_to_event_original
+      target_field: event.original
+      ignore_missing: true
+      description: Renames the original `message` field to `event.original` to store a copy of the original message. The `event.original` field is not touched if the document already has one; it may happen when Logstash sends the document.
+      if: ctx.event?.original == null
+
+  - remove:
+      field: message
+      tag: remove_message
+      ignore_missing: true
+      description: The `message` field is no longer required if the document has an `event.original` field.
+      if: ctx.event?.original != null
+  - json:
+      field: event.original
+      tag: json_event_original
+      target_field: tencent_cloud.audit
+      if: ctx.event?.original != null
+      on_failure:
+      - append:
+          field: error.message
+          value: 'Processor {{{_ingest.on_failure_processor_type}}} with tag {{{_ingest.on_failure_processor_tag}}} in pipeline {{{_ingest.on_failure_pipeline}}} failed with message: {{{_ingest.on_failure_message}}}'
+      description: Parses the `event.original` field as JSON and stores the result in the `audit` field. If the JSON parsing fails, an error message is appended to the `error.message` field.
+  - rename:
+      ignore_missing: true
+      field: tencent_cloud.audit.userIdentity.accountId
+      target_field: cloud.account.id
+  - rename:
+      ignore_missing: true
+      field: tencent_cloud.audit.userIdentity.userName
+      target_field: user.name
+  - rename:
+      ignore_missing: true
+      ignore_failure: true
+      field: tencent_cloud.audit.eventRegion
+      target_field: cloud.region
+  - rename:
+      ignore_missing: true
+      field: tencent_cloud.audit.errorCode
+      target_field: error.code
+  - rename:
+      ignore_missing: true
+      field: tencent_cloud.audit.requestID
+      target_field: event.id
+  - rename:
+      ignore_missing: true
+      field: tencent_cloud.audit.httpMethod
+      target_field: http.request.method
+  - rename:
+      ignore_missing: true
+      field: tencent_cloud.audit.userAgent
+      target_field: user_agent.original
+  - script:
+      tag: script_convert_array_to_string
+      description: Ensures the date processor does not receive an array value.
+      lang: painless
+      source: >
+        if (ctx.tencent_cloud.audit?.eventTime !=
+        null &&
+            ctx.tencent_cloud.audit.eventTime instanceof ArrayList){
+            ctx.tencent_cloud.audit.eventTime = ctx.tencent_cloud.audit.eventTime[0];
+        }
+  - date:
+      if: ctx.tencent_cloud.audit?.eventTime != null
```

```suggestion
      if: ctx.tencent_cloud?.audit?.eventTime != null
```
Please update for all null-checks in all processors.